gammapy/data/data_store.py

```diff
@@ -335,3 +392,216 @@ def check_observations(self):
         for obs_id in self.data_store.obs_table["OBS_ID"]:
             obs = self.data_store.obs(obs_id)
             yield from ObservationChecker(obs).run()
+
+
+class DataStoreMaker:
```

Comment:
Well, a DataStore really is just the two index tables.

Added this method to make it clear:
```
data_store = DataStoreMaker(paths).run()
```

I realise the code will need some further work / refactoring. E.g. `CalDBIRF` and `load_cta_irf` should be merged / improved. The functionality on `DataStoreMaker` might partially or completely be moved to `HDUIndexTable` and `ObservationTable` (if we keep that class).

For now I keep like this, because its not clear to me what better code org makes sense yet.